api/src/mcp/tools/assets.ts

```diff
@@ -0,0 +1,39 @@
+import { z } from 'zod';
+import { AssetsService } from '../../services/assets.js';
+import { defineTool } from '../define.js';
+import prompts from './prompts/index.js';
+
+const AssetsValidateSchema = z.strictObject({
+	id: z.string(),
+});
+
+const AssetsInputSchema = z.object({
+	id: z.string(),
+});
+
+export const assets = defineTool<z.infer<typeof AssetsValidateSchema>>({
+	name: 'assets',
+	description: prompts.assets,
+	inputSchema: AssetsInputSchema,
+	validateSchema: AssetsValidateSchema,
+	async handler({ args, schema, accountability }) {
+		const assetsService = new AssetsService({
+			accountability,
+			schema,
+		});
+
+		const asset = await assetsService.getAsset(args.id);
+
+		const chunks = [];
+
+		for await (const chunk of asset.stream) {
+			chunks.push(Buffer.from(chunk));
+		}
+
+		return {
+			type: 'image',
+			data: Buffer.concat(chunks).toString('base64'),
+			mimeType: 'image/png',
+		};
```

Comment:
Because assets can be any file type should we:
- check the file mime type for images to ensure we are actually returning an image and throw an error for other types?
```suggestion
        if (!asset.file.type.startsWith('image/')) throw Error();
		return {
			type: 'image',
			data: Buffer.concat(chunks).toString('base64'),
			mimeType: asset.file.type,
		};
```

- support other types which are supported by MCP `text, image, audio`?
```suggestion
		return {
			type: 'image' | 'audio' | 'text', // determine based on file type
			data: Buffer.concat(chunks).toString('base64'),
			mimeType: asset.file.type,
		};
```

Reply:
Added audio support and fixed mimetype via c0ce853b349b54a5a0a34e65dd93c43c7beeb4c4. Restricted return to image or audio as [MCP spec does not currently have a format for returning files as data](https://modelcontextprotocol.io/specification/2025-06-18/server/tools#data-types)